sql/src/main/java/io/crate/planner/operators/Rename.java

```diff
@@ -80,6 +84,58 @@ public RelationName name() {
         return outputs;
     }
 
+    @Override
+    public LogicalPlan pruneOutputsExcept(Collection<Symbol> outputsToKeep) {
+        ArrayList<Symbol> mappedToKeep = new ArrayList<>();
+        /* outputsToKeep may be scalars on top of the Rename result or subscripts
+         * Eval: x > 1, obj['x']
+         *  └ Rename[x, obj]
+         *      └ Collect [x, obj]
+         *
+         * TODO: `obj['x']` can be a ScopedSymbol instead of a subscript function. Can we change that?
```

Comment:
can you elaborate a bit here? if we use a ScopedSymbol instead of a subscript function, we basically do not need the logic in the loop bellow? how can a subscript function be mapped to scoped symbol?

Reply:
Yeah the comment is a bit vague because even if it is a subscript function it will be wrapped in a `ScopedSymbol`. The point is more about whether the `path` component of the `ColumnIdent` is used vs. having a real `Function`. In the latter case the `FieldsVisitor` approach would be enough and the `getRoot()` part further below wouldn't be necessary.

I'll adapt the comment or figure out a better solution.